utilities/utils/response_unix.go

```diff
@@ -4,8 +4,65 @@ package utils
 
 import (
 	"github.com/docker/docker/api/types"
+	"github.com/patrickmn/go-cache"
+	"github.com/rancher/agent/utilities/config"
+	"github.com/rancher/agent/utilities/constants"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+	"time"
 )
 
-func getIP(inspect types.ContainerJSON) string {
-	return inspect.NetworkSettings.IPAddress
+const (
+	cniLabels = "io.rancher.network.cni"
+	linkName  = "eth0"
+)
+
+func getIP(inspect types.ContainerJSON, c *cache.Cache) (string, error) {
+	if val, ok := inspect.Config.Labels[cniLabels]; ok && val == "true" && c != nil {
+		done := make(chan bool, 1)
+		ip := ""
+		go func() {
+			initTime := time.Duration(250) * time.Millisecond
+			maxTime := time.Duration(2) * time.Second
+			for {
```

Comment:
If something keeps failing (like the container doesn't exist) will this just loop forever?  It need to abort at some point.